### PR TITLE
Improvements in UAVCAN GNSS bridge

### PR DESCRIPTION
**Pull request by pavel-kirienko**
- Added support for the [new Fix2 data type](https://github.com/UAVCAN/dsdl/pull/10). The bridge automatically switches from Fix to Fix2 depending on what is available on the bus.

- Fixed time base conversion. PX4 uses UTC internally, whereas UAVCAN GNSS fix message may contain either UTC, GPS, or TAI time. Necessary conversion code has been added.

- If there are no UAVCAN nodes that provide GNSS position data, the bridge will be broadcasting this data itself. Broadcasting stops automatically if another node that provides GNSS data is detected.

Changes have been tested on Pixracer.

### CODE REVIEW

File: src/modules/uavcan/sensors/gnss.cpp [hunk 1]
```diff
@@ -43,25 +43,48 @@
 #include <drivers/drv_hrt.h>
 #include <systemlib/err.h>
 #include <mathlib/mathlib.h>
+#include <algorithm>
```
Comment (bkueng) [inline src/modules/uavcan/sensors/gnss.cpp]: This cannot be used on NuttX

File: src/modules/uavcan/sensors/gnss.cpp [hunk 2]
```diff
@@ -43,25 +43,48 @@
...
+{
+	(void) orb_unsubscribe(_orb_sub_gnss);
+}
+
 int UavcanGnssBridge::init()
 {
-	int res = _sub_fix.start(FixCbBinder(this, &UavcanGnssBridge::gnss_fix_sub_cb));
+	int res = _pub_fix2.init(uavcan::TransferPriority::MiddleLower);
+	if (res < 0) {
+		warnx("GNSS fix2 pub failed %i", res);
```
Comment (bkueng) [inline src/modules/uavcan/sensors/gnss.cpp]: Please use `PX4_{INFO,WARN,ERR}`, `warnx` is deprecated.

File: src/modules/uavcan/sensors/gnss.cpp (hunk 1 above)
Comment (pavel-kirienko) [inline src/modules/uavcan/sensors/gnss.cpp]: Fixed.

File: src/modules/uavcan/sensors/gnss.cpp (hunk 2 above)
Comment (pavel-kirienko) [inline src/modules/uavcan/sensors/gnss.cpp]: Fixed.

**Pull request merged**